### ISSUE TITLE
Deploy DataPack- Omnscript & IP DataPack Json File jumbled.

### ISSUE THREAD

**Issue by a470836-abloomfield** [Codeneos/vlocode#396]
When an Omniscript or IP is deployed, the generated Datapack.json file elements is rearranged or jumbled.   When looking at it from inside omnistudio it seems the elements are in the correct order but when you preview or run the OS or IP the elements are out of order.  If you try to export it after deploying it, the elements come back in the  in correct order.

Editor used: VS Code.
Version:  v0.18.18

**Comment by Codeneos** [comment]
This happens when the level and order properties of the elements are missing. Under normal circumstances these are re-calculated generated during a deployment by looking at the order of the elements.  

I suspect that this is not happening which is why the elements are jumbled.  

If possible can you share with me the datapack of the OmniScript, you can mask any customer sensitive data as I am only interested in the structure of the datapack so I can reproduce it and confirm the root cause.

**Comment by Codeneos** [comment]
@a470836-abloomfield 

This should be fixed in the next release; order and level are now re-calculated at deploy time. See sommit [c9c08f1](https://github.com/Codeneos/vlocode/commit/c9c08f1774fec7cda1c157fbf03c8e9e5fba8768) for the details.

**Comment by a470836-abloomfield** [comment]
@Codeneos  I would love to contribute to this project. Let me know if you need any help.

**Comment by a470836-abloomfield** [comment]
> @a470836-abloomfield this should be fixed in the next release; order and level are now re-calculated at deploy time.
> 
> See sommit [c9c08f1](https://github.com/Codeneos/vlocode/commit/c9c08f1774fec7cda1c157fbf03c8e9e5fba8768#diff-9d0391866c8dc23c28f31de354cf0157ce8cda515caba8fa0f79ee9b30fc80e5R70) for the details.

I am curious because it is not happening to all users though. Just some

**Comment by Codeneos** [comment]
When you export an OmniScript or IP in the past it included a `Level__c` and `Order__c` property which determin the position of the element in the script tree,

Later versions the datapack strip these properties while exporting and automatically recalculate them while importing based on the order of the elements in the datapack. The part responsible for recalculating these properties was not yet ported to vlocode's **direct** deploy strategy and only available in the **compatibility** strategy.

If you see jumbled up elements in the script tree it is definitely related to either of these properties not being deployed correctly.

Another solution is to make sure that `Level__c` and `Order__c` properties are exported, this can be done by using an override definition which to instruct the export to not filter these properties. This way you can be sure the elements are deployed in the correct order.

Sample:
```yaml
OverrideSettings:
  SObjects:
    vlocity_namespace__Element__c:
      BuildOnlyFields: {}
      FilterFields:
      - vlocity_namespace__OmniScriptVersion__c
```

**Comment by Codeneos** [comment]
The above should be fixed in the 0.19.x branch; if the issue still occurs please re-open the issue.
